File: Crash/Models/ContaPagar.cs
using System.ComponentModel; 
using System.ComponentModel.DataAnnotations;

namespace Crash.Models;

public class ContaPagar
{
    public int Id { get; set; }
    [Required]
    public required string Descricao { get; set; } = string.Empty;
    [Required]
    public required decimal Valor { get; set; }
    [Required]
    public required DateTime DataVencimento { get; set; }

    [DefaultValue(false)]

    public bool Pago { get; set; } = false;

    // Relacionamentos Obrigatórios (Base)
    [Required]
    public required int FornecedorId { get; set; }
    public Fornecedor? Fornecedor { get; set; }
    [Required]
    public required int ContaContabilId { get; set; }
    public ContaContabil? ContaContabil { get; set; }
    [Required]
    public required int CentroCustoId { get; set; }
    public CentroCusto? CentroCusto { get; set; }

    // Relacionamento com Conta Bancária (Preenchido só quando ocorrer a baixa)
    public int? ContaBancariaId { get; set; }
    public ContaBancaria? ContaBancaria { get; set; }
}

File: FrontendCrash/src/models/ContaPagar.ts
import type Fornecedor from "./Fornecedor";
import type ContaContabil from "./ContaContabil";
import type CentroCusto from "./CentroCusto";
import type ContaBancaria from "./ContaBancaria";

// Interface que representa o modelo ContaPagar (título a pagar)
// Possui relacionamentos com Fornecedor, ContaContabil, CentroCusto e ContaBancaria
export default interface ContaPagar {
    id?: number;
    descricao: string;
    valor: number;
    dataVencimento: string;
    pago?: boolean;
    // FKs — ids das entidades relacionadas (enviados no POST/PUT)
    fornecedorId: number;
    contaContabilId: number;
    centroCustoId: number;
    contaBancariaId?: number;
    // Objetos de navegação — retornados pelo GET com Include
    fornecedor?: Fornecedor;
    contaContabil?: ContaContabil;
    centroCusto?: CentroCusto;
    contaBancaria?: ContaBancaria;
}
